### ISSUE TITLE
Possible error in Op_RGB_to_YCbCr<Pixel>::convert_colorspace

### ISSUE THREAD

**Issue by chris0e3** [strukturag/libheif#445]
The following code looks wrong.
In function Op_RGB_to_YCbCr<Pixel>::convert_colorspace in heif_colorconversion.cc, line 728:
```
        if (!full_range_flag) {
          cb = (cb * 224) / 256;
          cr = (cb * 224) / 256;    // ← ????
        }
```
I think it should probably be:
```
        if (!full_range_flag) {
          cb = (cb * 224) / 256;
          cr = (cr * 224) / 256;    // ← change here
        }
```

Additionally, I was wondering about all the `… / 256` occurrences in this function - as it is templated to support 8-bit or 16-bit pixels.  Is this correct for 16-bit?

**Comment by silverbacknet** [comment]
This is not perfectly correct for higher bit depths, but it's so close it doesn't matter. In 8-bit, 240 is peak and (255 * 224 / 256 + 16) gets you 239.125, but in 10-bit, 960 is peak and (255 * 224 / 256 + 16) << 2 is 959.12, a closer approximation. This may actually *just work*™ due to overflows in the initial matrix transformation not being clamped early.

Fortunately, limited range is rarely used in images, I wish HD had done away with it for video.

**Comment by farindk** [comment]
@chris0e3 Thank you for spotting the typo in the color conversion. I have fixed it (9e8030c).

Concerning the 224/256 factor: this actually correct for both 8-bit and HDR. I'm following the specification in H.273 (eqs. 24 and 25). Just consider 224/256 as a factor. It originates from 8-bit (hence the 256), but it is the same scaling factor for HDR.

@silverbacknet Yes, I also wished the limited range technical debt was finally deprecated.
@silverbacknet BTW: there is no +16 for the color channels. Hence, it matches perfectly.

**Comment by silverbacknet** [comment]
That's right, you start from float input already, not existing truncated values, so they go the full range. My bad, I'm sorry about that.
